substrate/client/sysinfo/src/sysinfo.rs

```diff
@@ -375,6 +388,48 @@ pub fn benchmark_cpu(limit: ExecutionLimit) -> Throughput {
 		.expect("benchmark cannot fail; qed")
 }
 
+// This benchmarks the entire CPU speed as measured by calculating BLAKE2b-256 hashes, in bytes per
+// second. It spawns multiple threads to measure the throughput of the entire CPU and averages the
+// score obtained by each thread. If we have at least `refhw_num_cores` available then the
+// average throughput should be relatively close to the single core performance as measured by
+// `benchmark_cpu`.
+pub fn benchmark_cpu_parallelism(limit: ExecutionLimit, refhw_num_cores: usize) -> Throughput {
+	const SIZE: usize = 32 * 1024;
+
+	let ready_to_run_benchmark = Arc::new(Barrier::new(refhw_num_cores));
+	let mut benchmark_threads = Vec::new();
+
+	// Spawn a thread for each expected core and average the throughput for each of them.
+	for _ in 0..refhw_num_cores {
+		let ready_to_run_benchmark = ready_to_run_benchmark.clone();
+
+		let handle = std::thread::spawn(move || {
+			let mut buffer = Vec::new();
+			buffer.resize(SIZE, 0x66);
+			let mut hash = Default::default();
+
+			let run = || -> Result<(), ()> {
```

Comment:
This is copy&pasted from the function above?  

I guess it could either reuse that, or delete the single-threaded function and just call this one with a single core?

Reply:
Done.